terraform/cos/applications.tf

```diff
@@ -80,20 +80,28 @@ module "mimir" {
   s3_integrator_revision           = var.s3_integrator.revision
   s3_integrator_storage_directives = var.s3_integrator.storage_directives
   s3_integrator_units              = var.s3_integrator.units
-  coordinator_config               = var.mimir_coordinator.config
-  coordinator_constraints          = var.mimir_coordinator.constraints
-  coordinator_revision             = var.mimir_coordinator.revision
-  coordinator_storage_directives   = var.mimir_coordinator.storage_directives
-  coordinator_units                = var.mimir_coordinator.units
-  backend_config                   = var.mimir_worker.backend_config
-  read_config                      = var.mimir_worker.read_config
-  write_config                     = var.mimir_worker.write_config
-  worker_constraints               = var.mimir_worker.constraints
-  worker_revision                  = var.mimir_worker.revision
-  worker_storage_directives        = var.mimir_worker.storage_directives
-  backend_units                    = var.mimir_worker.backend_units
-  read_units                       = var.mimir_worker.read_units
-  write_units                      = var.mimir_worker.write_units
+  coordinator_config = merge(
+    var.mimir_coordinator.config,
+    # enable exemplar storage (required for metrics-to-traces).
+    # This config option is not supported in track `1`, so we'll set it only
+    # for newer tracks to maintain backward compatibility.
+    can(regex("^1/", var.channel)) ? {} : {
+      "max_global_exemplars_per_user" = "100000"
+    }
```

Comment:
Terraform plans also follow the track notation. This will go into track 2, so this guard is probably unnecessary.
To my understanding, the `var.channel` is mostly to switch between edge..stable. If a track/3 terraform module happens to work with track/2 charms is a bonus, not a design target.
But maybe I'm missing something.
In any case, this seems simple enough :)